### DISCUSSION TITLE
OrchardCore - Problem when 2 webprojects have same controller name and action name

### DISCUSSION THREAD

**Discussion by nsathiyamoorthi** [OrchardCMS/OrchardCore#7155] [Q&A]
We are planning to implement modular concept in our application using OrchardCore. While doing POC we ran into below problem. 

We have 2 web projects 

- Shell Project, contains the layout, shared static assets (css,js) and common home page (Home Controller with action Index)

- Module1,This also has a controller called "Home" with action "Index"

Issue is even when i call the route https://localhost:44335/Module1/Home/Index , it is still redirecting me to https://localhost:44335:/home/index

But call to my other controller https://localhost:44335:Module1/Management/Index is working as expected. only if both have the same Controller and Action, then shell project takes preference automatically. 

Startup in Shell
![image](https://user-images.githubusercontent.com/26634613/94476409-71c30980-019e-11eb-8f20-982c6e43ee30.png)

Startup in Module1 
![image](https://user-images.githubusercontent.com/26634613/94476345-59eb8580-019e-11eb-875e-8eb087ed5f90.png)

How do i force the system to redirect user to corresponding page even if 2 projects have the same controller name?

**Comment by Skrypt** [comment]
Try using MapAreaControllerRoute in both projects.

**Reply by nsathiyamoorthi** [reply]
we tried that also. still having the same problem

**Reply by Skrypt** [reply]
Have you set the AreaName of the route in both? Because I think that the issue is that you are redefining the default MVC home route of your website on the first route. So you have a route override/collision in your code. When you do a module normally you define the areaName of the module for this to map to the proper path.

**Comment by jtkech** [comment]
Don't use `UseEnpoints()` from your app it is done for you, you could but not for what you are trying to do.

In place, use one of our `OrchardCoreBuilder` helpers to register / configure things in a tenant container, as it was done through a module startup, e.g.

            services.AddOrchardCms()
                .Configure((tenantAppBuilder, tenantEndpointRouteBuilder) =>
                {
                    tenantEndpointRouteBuilder.MapAreaControllerRoute(
                        name: "WhatYouWant",
                        areaName: "TheAssemblyNameOfYourApp",
                        // and so on
                        );
                })
                ;

And as @Skrypt said, always use an area, it is usually a module name, but when defined from the app it is the app name. Note: The application (or part of it) can behave as a module, but the related things need to be registered / configured in a tenant container as it is done through a module startup, this is the goal of our `OrchardCorebuilder` helpers.
